kong/llm/drivers/shared.lua

```diff
@@ -229,6 +242,75 @@ local function handle_stream_event(event_table, model_info, route_type)
   end
 end
 
+---
+-- Manages cloud SDKs, for using "workload identity" authentications,
+-- that are tied to this specific plugin in-memory.
+--
+-- This allows users to run different authentication configurations
+-- between different AI Plugins.
+--
+-- @param {table} this_cache self - stores all the SDK instances
+-- @param {table} plugin_config the configuration to cache against and also provide SDK settings with
+-- @return {table} self
+_M.cloud_identity_function = function(this_cache, plugin_config)
+  if plugin_config.model.provider == "gemini" and
+      plugin_config.auth and
+      plugin_config.auth.gcp_use_service_account then
+
+    ngx.log(ngx.NOTICE, "loading gcp sdk for plugin ", kong.plugin.get_id())
+
+    local service_account_json = (plugin_config.auth and plugin_config.auth.gcp_service_account_json) or GCP_SERVICE_ACCOUNT
```

Comment:
Inside this `if` branch, `service_account_json` I think can always get its value from `plugin_config.auth.gcp_service_account_json`. Should we change it to:

```
local service_account_json = plugin_config.auth.gcp_service_account_json
```

Reply:
Non, there are two places it can come from: the `plugin_config.auth.gcp_service_account_json` can be unset, in which case it will default environment variable GCP_SERVICE_ACCOUNT